pkg/dtest/subprocess.go

```diff
@@ -0,0 +1,105 @@
+package dtest
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+)
+
+func getFunctionName(i interface{}) string {
+	return runtime.FuncForPC(reflect.ValueOf(i).Pointer()).Name()
+}
+
+type sub struct {
+	sudo      bool
+	functions map[string]func()
+}
+
+func (s *sub) _make(sudo bool, f func()) *exec.Cmd {
+	name := getFunctionName(f)
+	s.functions[name] = f
+	var cmd *exec.Cmd
+	if sudo {
+		cmd = exec.Command("sudo", "-E", os.Args[0])
+		s.sudo = true
+	} else {
+		cmd = exec.Command(os.Args[0])
+	}
+	cmd.Env = append(os.Environ(), fmt.Sprintf("BE_SUBPROCESS=%s", name))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd
+}
+
+func (s *sub) Make(f func()) *exec.Cmd {
+	return s._make(false, f)
+}
+
+func (s *sub) MakeSudo(f func()) *exec.Cmd {
+	return s._make(true, f)
+}
+
+func (s *sub) Enable() {
+	name := os.Getenv("BE_SUBPROCESS")
```

Comment:
Since `go test` compiled binaries use the `flag` package to parse arguments, you can add in your own flags to the global `flag.CommandLine`, if it makes more sense to use a flag instead of an environment variable.

Reply:
isn't it better to avoid depending on an implementation detail of go test?

Reply:
or is that behavior guaranteed?

Reply:
I think it's guaranteed.

Reply:
done